src/hooks/screen/useChatScreenHook.ts

```diff
@@ -0,0 +1,153 @@
+    iteration = 0,
+    sendingChatSchema: ChatSchema | null = null
+  ) => {
+    if (iteration > 5) {
+      SimpleToast.show("Can't send message, please check your connection");
+      return;
+    }
+
+    let currentChatSchema = sendingChatSchema;
+    let userId = await getUserId();
+    const myAnonymousId = await getAnonymousUserId();
+
+    if (type === 'ANONYMOUS') {
+      userId = myAnonymousId;
+    }
+    try {
+      const randomId = uuid();
+
+      if (currentChatSchema === null) {
+        currentChatSchema = await ChatSchema.generateSendingChat(
+          randomId,
+          userId,
+          selectedChannel?.id,
+          message,
+          localDb
+        );
+        currentChatSchema.save(localDb);
+        refresh('chat');
+        refresh('channelList');
+      }
+      let channelType = CHANNEL_TYPE_PERSONAL;
+      if (
+        selectedChannel?.channelType?.toLowerCase() === 'group' ||
+        selectedChannel?.rawJson?.channel?.type === 'group'
+      ) {
+        channelType = CHANNEL_TYPE_GROUP;
+      }
+
+      if (selectedChannel?.rawJson?.channel?.channel_type === 4) {
+        channelType = CHANNEL_TYPE_ANONYMOUS;
+      }
+
+      let response;
+      if (type === 'ANONYMOUS') {
+        response = await AnonymousMessageRepo.sendAnonymousMessage(selectedChannel?.id, message);
+      } else {
+        response = await SignedMessageRepo.sendSignedMessage(
+          selectedChannel?.id,
+          message,
+          channelType
+        );
+      }
+      await currentChatSchema.updateChatSentStatus(localDb, response);
+      refresh('chat');
+      refresh('channelList');
+    } catch (e) {
+      if (e?.response?.data?.status === 'Channel is blocked') return;
+
+      setTimeout(() => {
+        sendChat(message, iteration + 1, currentChatSchema).catch((sendChatError) => {
+          console.log(sendChatError);
+        });
+      }, 1000);
+    }
+  };
+
+  const handleUserName = (item): string => {
+    if (item?.user?.username !== 'AnonymousUser') {
+      return item?.user?.username;
+    }
+    return `Anonymous ${anon_user_info_emoji_name}`;
+  };
+
+  const updateChatContinuity = (chatsData: ChatSchema[]) => {
+    const updatedChats = chatsData.map((currentChat, currentIndex) => {
+      const previousChat = chatsData[currentIndex + 1];
+
+      if (previousChat) {
+        if (
+          currentChat?.userId === previousChat?.userId &&
+          (previousChat?.rawJson?.isSystem || previousChat?.rawJson?.type === 'system')
+        ) {
+          currentChat.isContinuous = false;
+        }
+      }
+      return currentChat;
+    });
+
+    return updatedChats;
+  };
+
+  React.useEffect(() => {
+    initChatData();
+  }, [localDb, chat, selectedChannel]);
+
+  return {
+    chats,
+    selectedChannel,
+    goBackFromChatScreen,
+    goToChatInfoScreen,
+    sendChat,
+    handleUserName,
+    updateChatContinuity
+  };
+}
+
+export default useChatScreenHook;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the useEffect hook, you're calling `initChatData()` without wrapping it in an async function. This can lead to warnings about promises not being handled correctly. Consider wrapping it in an async function.
```diff
- React.useEffect(() => {
-   initChatData();
- }, [localDb, chat, selectedChannel]);
+ React.useEffect(() => {
+   const fetchData = async () => {
+     await initChatData();
+   };
+   fetchData();
+ }, [localDb, chat, selectedChannel]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->